### ISSUE TITLE
Memory leak of ComposeWindow/ComposeLayer after `window.dispose`

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#1688]
Compose 1.0.1, OpenJDK 11/15, Windows 11

```
import androidx.compose.ui.awt.ComposeWindow
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.swing.Swing

fun main() {
    runBlocking(Dispatchers.Swing) {
        repeat(10) {
            val c = ComposeWindow()
            c.setContent {
            }
            c.isVisible = true
            delay(100)
            c.dispose()
        }
        delay(320000)
    }
}
```

![image](https://user-images.githubusercontent.com/5963351/149167835-19422cec-1130-443b-b19a-964b4dee78dc.png)

**Comment by igordmn** [comment]
Pure Swing leak:
```
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.swing.Swing
import javax.swing.JFrame

fun main() {
    runBlocking(Dispatchers.Swing) {
        repeat(10) {
            val c = JFrame()
            c.isVisible = true
            c.contentPane.requestFocus()
            delay(100)
            c.dispose()
        }
        delay(320000)
    }
}
```

**Comment by igordmn** [comment]
Linux, JBR 11:
![image](https://user-images.githubusercontent.com/5963351/149176189-b18756da-c389-40e5-8a47-31b0b5e1ee3a.png)

**Comment by igordmn** [comment]
Will be fixed in 1.0.2

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-1688">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
